### ISSUE TITLE
Ligatures=TeX has no influence with certain fonts

### ISSUE THREAD

**Issue by simifilm** [wspr/fontspec#237]
Consider the following MWE (with `lualatex`):

```
\documentclass{article}
\usepackage{fontspec}
\setmainfont{Palatino LT Std}[
Ligatures = TeX,
Extension = .otf,
UprightFont = *-Roman,
ItalicFont     = *-Italic,
BoldFont       = *-Bold,
BoldItalicFont = *-BoldItalic,
]

\begin{document}
--
 \end{document}
```

Although `Ligatures=TeX` is set, the two dashes are not rendered as – (em dash). 

I've seen this with several fonts, but I could not figure out any systematics behind it. What I don't get is that in my understanding `Ligatures=TeX` is not a feature of the font but only concerns the mapping of -- to –.

I realize that this MWE is not ideal since it relies on a proprietary font, but if needed I can provide the font.

**Comment by wspr** [comment]
The best way to check where this this is coming from would be with a plain LuaTeX document, such as:

    \input luaotfload.sty\relax

    \font\x="file:texgyrepagella-regular.otf:+tlig" at 10pt\relax
    \x abc---xyz

    \bye

It’s improbable that this is a `fontspec` issue…

**Comment by simifilm** [comment]
This example does actually produce an em dash.

**Comment by wspr** [comment]
With your font or just the example given? If you get an em-dash there where in fontspec you don’t, can you check what the log file is saying about loading the font in the fontspec case? E.g., you’ll get something like this:

    .................................................
    . fontspec info: "defining-font"
    . 
    . Font family 'texgyrepagella-regular.otf(0)' created for font
    . 'texgyrepagella-regular.otf' with options [Ligatures=TeX].
    . 
    . This font family consists of the following shapes:
    . 
    . * 'normal' with NFSS spec.:
    . 
    . <->"[texgyrepagella-regular]:mode=node;script=latn;language=DFLT;+tlig;+trep;"
    
    . 
    . * 'small caps' with NFSS spec.:
    . 
    . <->"[texgyrepagella-regular]:mode=node;script=latn;language=DFLT;+tlig;+trep;+
    smcp;"
    . and font adjustment code:
    . 
    ………………………………………….

(there’ll be differences in fontspec version in terms of what the exact text is, but the important part is what comes after the `<→` bit.

**Comment by simifilm** [comment]
Sorry, I'm dumb, with my font, I get the hyphens.

**Comment by wspr** [comment]
In that case it's not fontspec's fault `:)` probably best to report to the luaotfload repository.

**Comment by simifilm** [comment]
For what it's worth, this seems to have been fixed with `luaotfload` 2.7.

**Comment by wspr** [comment]
Thanks for the update.
